keychain-sdk/internal/tracker/tracker.go

```diff
@@ -1,35 +1,72 @@
 package tracker
 
 import (
+	"errors"
 	"sync"
 )
 
+type Action int
+
+const (
+	ActionSkip Action = iota
+	ActionProcess
+)
+
+type stringSet map[string]struct{}
+
+// add safely adds a string to the set.
```

Comment:
there's no "safety" here, i suggest we rephrase it like this:

```suggestion
// add adds a string to the set and returns true if the key was already present.
```